### PR TITLE
docs: add push check to /pre-pr and branch protection rule

### PR DESCRIPTION
**Pull request by joshsmithxrm**
## Summary

- Add explicit step 6 to `/pre-pr` command to verify changes are pushed before creating PR
- Add "Commit directly to main" to NEVER section in CLAUDE.md

## Background

Discovered during session retrospective that PR creation failed because commits weren't pushed to remote. This is a common oversight that the `/pre-pr` command should catch.

Also adds the branch protection rule to SDK's CLAUDE.md since the workspace CLAUDE.md had it but SDK-specific didn't.

## Test plan

- [ ] Verify `/pre-pr` command includes push check step
- [ ] Verify CLAUDE.md NEVER section includes branch protection rule

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: .claude/commands/pre-pr.md [hunk 1]
```diff
@@ -102,8 +102,22 @@ Missing tests is a blocker. Writing tests now...
...
+### 6. Push Check
+
+Before creating PR, ensure changes are pushed:
+
+```bash
+# Check if ahead of remote
+git status
+
+# If ahead, push
+git push -u origin <branch-name>
```
[the remark this answers is not in the document]
Comment (joshsmithxrm) [reply .claude/commands/pre-pr.md]: Fixed in 05e67bd - Updated to use $(git branch --show-current) for copy-paste convenience.

File: .claude/commands/pre-pr.md [hunk 2]
```diff
@@ -102,8 +102,22 @@ Missing tests is a blocker. Writing tests now...
...
+```
+
+**The PR cannot be created if commits aren't pushed.** This is a common oversight.
```
[the remark this answers is not in the document]
Comment (joshsmithxrm) [reply .claude/commands/pre-pr.md]: Fixed in 05e67bd - Moved Push Check section to be #6 under Checks Performed, before Output.

**Comment by joshsmithxrm** [comment]
@gemini-code-assist review

File: .claude/commands/pre-pr.md [hunk 3]
```diff
@@ -79,6 +79,20 @@ public void Add_FirstProfile_SetsAsActive()
...
+### 6. Push Check
+
+Before creating PR, ensure changes are pushed:
+
+```bash
+# Check if ahead of remote
+git status
+
+# If ahead, push
+git push -u origin $(git branch --show-current)
```
[the remark this answers is not in the document]
Comment (joshsmithxrm) [reply .claude/commands/pre-pr.md]: Fixed in c886bd1 - Updated to use git rev-parse --abbrev-ref HEAD with quotes for better portability.

File: .github/workflows/test.yml [hunk 4]
```diff
@@ -3,8 +3,22 @@ name: Test
 on:
   push:
     branches: [main]
+    paths:
+      - 'src/**'
+      - 'tests/**'
+      - '**/*.csproj'
+      - '*.sln'
+      - 'Directory.Build.props'
+      - '.github/workflows/**'
```
[the remark this answers is not in the document]
Comment (joshsmithxrm) [reply .github/workflows/test.yml:12]: Declining - The broad pattern is intentional. When any workflow file changes, we want all workflows to run to validate the change doesn't break the build/test process. Using specific paths would create maintenance burden and miss edge cases.

File: .github/workflows/integration-tests.yml [hunk 5]
```diff
@@ -3,8 +3,22 @@ name: Integration Tests
 on:
   push:
     branches: [main]
+    paths:
+      - 'src/**'
+      - 'tests/**'
+      - '**/*.csproj'
+      - '*.sln'
+      - 'Directory.Build.props'
+      - '.github/workflows/**'
```
[the remark this answers is not in the document]
Comment (joshsmithxrm) [reply .github/workflows/integration-tests.yml:12]: Declining - Same rationale as above. The broad workflow pattern is intentional for safety.

File: .github/workflows/build.yml [hunk 6]
```diff
@@ -3,8 +3,22 @@ name: Build
 on:
   push:
     branches: [main]
+    paths:
+      - 'src/**'
+      - 'tests/**'
+      - '**/*.csproj'
+      - '*.sln'
+      - 'Directory.Build.props'
+      - '.github/workflows/**'
```
[the remark this answers is not in the document]
Comment (joshsmithxrm) [reply .github/workflows/build.yml:12]: Declining - Same rationale as above. The broad workflow pattern is intentional for safety.

File: .github/workflows/test.yml [hunk 7]
```diff
@@ -3,8 +3,22 @@ name: Test
...
   pull_request:
     branches: [main]
+    paths:
+      - 'src/**'
+      - 'tests/**'
+      - '**/*.csproj'
+      - '*.sln'
+      - 'Directory.Build.props'
+      - '.github/workflows/**'
```
[the remark this answers is not in the document]
Comment (joshsmithxrm) [reply .github/workflows/test.yml:21]: Declining - Same rationale as above. The broad workflow pattern is intentional for safety.

File: .github/workflows/integration-tests.yml [hunk 8]
```diff
@@ -3,8 +3,22 @@ name: Integration Tests
...
   pull_request:
     branches: [main]
+    paths:
+      - 'src/**'
+      - 'tests/**'
+      - '**/*.csproj'
+      - '*.sln'
+      - 'Directory.Build.props'
+      - '.github/workflows/**'
```
[the remark this answers is not in the document]
Comment (joshsmithxrm) [reply .github/workflows/integration-tests.yml:21]: Declining - Same rationale as above. The broad workflow pattern is intentional for safety.

File: .github/workflows/build.yml [hunk 9]
```diff
@@ -3,8 +3,22 @@ name: Build
...
   pull_request:
     branches: [main]
+    paths:
+      - 'src/**'
+      - 'tests/**'
+      - '**/*.csproj'
+      - '*.sln'
+      - 'Directory.Build.props'
+      - '.github/workflows/**'
```
[the remark this answers is not in the document]
Comment (joshsmithxrm) [reply .github/workflows/build.yml:21]: Declining - Same rationale as above. The broad workflow pattern is intentional for safety.

**Comment by joshsmithxrm** [comment]
@gemini-code-assist review

File: .claude/commands/pre-pr.md [hunk 10]
```diff
@@ -79,6 +79,20 @@ public void Add_FirstProfile_SetsAsActive()
 - File is already covered by integration tests in PPDS.LiveTests
 - User explicitly confirms "this doesn't need unit tests because [reason]"
 
+### 6. Push Check
+
+Before creating PR, ensure changes are pushed:
+
+```bash
+# Check if ahead of remote
+git status
```
[the remark this answers is not in the document]
Comment (joshsmithxrm) [reply .claude/commands/pre-pr.md:89]: Fixed in d99252f - Added git fetch before git status to ensure accurate remote comparison.

File: .claude/settings.json [hunk 11]
```diff
@@ -81,8 +81,8 @@
       "Bash(gh pr status:*)",
       "Bash(gh pr checkout:*)",
       "Bash(gh pr diff:*)",
-      "Bash(gh api repos/*/pulls/*/comments:*)",
-      "Bash(gh api repos/*/pulls/*/comments/*/replies:*)",
+      "Bash(gh api repos/*/ppds-sdk/pulls/*/comments:*)",
+      "Bash(gh api repos/*/ppds-sdk/pulls/*/comments/*/replies:*)",
```
[the remark this answers is not in the document]
Comment (joshsmithxrm) [reply .claude/settings.json]: Fixed in d99252f - Changed to 'Bash(gh api repos/:*)' which uses correct prefix match syntax. Claude Code only supports wildcards at the end with ':*' - middle wildcards are treated literally. See docs: https://docs.anthropic.com/en/docs/claude-code/settings#permission-settings
